gordon/record_checker.py

```diff
@@ -0,0 +1,112 @@
+import asyncio
+import logging
+import time
+
+from async_dns import types
+from async_dns.resolver import ProxyResolver
+
+
+class RecordChecker(object):
+    QUERYING_THRESHOLD = 60
+
+    def __init__(self, dns_ip):
+        """Setup RecordChecker object
+
+        Args:
+            dns_ip: DNS ip server to query
+        """
+        self._dns_ip = dns_ip
+        self._resolver = ProxyResolver()
+        self._resolver.set_proxies([self._dns_ip])
+
+    def _extract_record_data(self, record):
+        """Extract record data from a record dict
+
+        Args:
+            record (dict): DNS record as a dict
+        """
+        return record.values()
+
+    async def check_record(self, record):
+        """Asynchronously queries a DNS server for a given record
+            to check how long it takes the record to become available.
+
+        Args:
+            record (dict): DNS record as a dict
+            with record properties
+
+        """
+        start_time = time.time()
+
+        name, rr_data, r_type, ttl = self._extract_record_data(record)
+        r_type_code = types.get_code(r_type)
+
+        record_found_in_dns_srv = False
```

Comment:
I read this as a DNS SRV record, but I think mean server? Can I suggest naming this `resolvable_record` or something like this?